### DISCUSSION TITLE
How to specify string length

### DISCUSSION THREAD

**Discussion by webnoob** [prisma/prisma#1940] [Q&A] [answered]
I'm sure I must have missed some docs somewhere as this seems quite a trivial thing to be able to do but how can you specify the generated field length of a String type? I've looked in the scaler types and can't see the info.

For instance, the following model:
```
model Book {
  id Int  @default(autoincrement()) @id
  title String?
}
```
In MySQL DB the title field will be a `varchar(191)` datatype. How I can I set the length?

**Comment by pantharshit00** [comment, marked as answer]
This is something that will get covered via https://github.com/prisma/prisma2/issues/446

Right now you can get around this by setting the data type yourself manually in the database and then introspecting it using `prisma2 introspect`

**Reply by webnoob** [reply]
Thanks for the info.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
